pdb2sql/transform.py

```diff
@@ -111,51 +83,71 @@ def rot_xyz_around_axis(xyz, axis, angle, center=None):
                          ct + uz**2 * (1 - ct)]])
 
     # apply the rotation
-    return np.dot(rot_mat, (xyz - center).T).T + center
+    return rotation_matrix(xyz, rot_mat, center)
 
+########################################################################
+# Rotation using Euler anlges
+# see https://en.wikipedia.org/wiki/Rotation_matrix#General_rotations
+########################################################################
 
-def _rotation_euler(xyz, alpha, beta, gamma):
+def rot_euler(db, alpha, beta, gamma, **kwargs):
+    """Rotate molecule from Euler rotation axis.
+
+    Args:
+        alpha (float): angle of rotation around the x axis
+        beta (float): angle of rotation around the y axis
+        gamma (float): angle of rotation around the z axis
+        **kwargs: keyword argument to select the atoms.
+            See pdb2sql.get()
+    """
+    xyz = _get_xyz(db, **kwargs)
+    xyz = rotation_euler(xyz, alpha, beta, gamma)
+    _update(db, xyz, **kwargs)
+
+def rotation_euler(xyz, alpha, beta, gamma, center=None):
 
     # precomte the trig
     ca, sa = np.cos(alpha), np.sin(alpha)
     cb, sb = np.cos(beta), np.sin(beta)
     cg, sg = np.cos(gamma), np.sin(gamma)
 
-    # get the center of the molecule
-    xyz0 = np.mean(xyz, 0)
-
     # rotation matrices
     rx = np.array([[1, 0, 0], [0, ca, -sa], [0, sa, ca]])
     ry = np.array([[cb, 0, sb], [0, 1, 0], [-sb, 0, cb]])
-    rz = np.array([[cg, -sg, 0], [sg, cs, 0], [0, 0, 1]])
+    rz = np.array([[cg, -sg, 0], [sg, cg, 0], [0, 0, 1]])
 
-    rot_mat = np.dot(rz, np.dot(ry, rz))
+    # get rotation matrix
+    rot_mat = np.dot(rz, np.dot(ry, rx))
 
     # apply the rotation
-    return np.dot(rot_mat, (xyz - xyz0).T).T + xyz0
+    return rotation_matrix(xyz, rot_mat, center)
 
+########################################################################
+# Rotation using provided rotation matrix
+########################################################################
 
-def rotation_matrix(xyz, rot_mat, center=True):
-    if center:
-        xyz0 = np.mean(xyz)
-        return np.dot(rot_mat, (xyz - xyz0).T).T + xyz0
-    else:
-        return np.dot(rot_mat, (xyz).T).T
+def rot_mat(db, mat, **kwargs):
+    """Rotate molecule from a rotation matrix.
 
+    Args:
+        mat (np.array): 3x3 rotation matrix
+        **kwargs: keyword argument to select the atoms.
+            See pdb2sql.get()
+    """
+    xyz = _get_xyz(db, **kwargs)
+    xyz = rotation_matrix(xyz, mat)
+    _update(db, xyz, **kwargs)
+
+def rotation_matrix(xyz, rot_mat, center=None):
```

Comment:
that function name implies it returns the rotation matrix. I would change it to  `rotate` or `apply_rotation`.

Reply:
Good idea! fixed.